influxdb/src/main/scala/akka/stream/alpakka/influxdb/impl/InfluxDBFlowStage.scala

```diff
@@ -58,28 +61,70 @@ private[influxdb] class InfluxDBFlowStage[T, C](
   final class InfluxDBRecordLogic extends InfluxDBLogic {
 
     override protected def write(messages: immutable.Seq[InfluxDBWriteMessage[T, C]]): Unit =
-      messages.foreach {
-        case InfluxDBWriteMessage(point: Point, _) => {
-          influxDB.write(point)
+      messages
+        .filter {
+          case InfluxDBWriteMessage(point: Point, _, _, _) => {
+            true
+          }
+          case InfluxDBWriteMessage(others: AnyRef, _, _, _) => {
+            failStage(new RuntimeException(s"unexpected type Point or annotated with Measurement required"))
+            false
+          }
         }
-        case InfluxDBWriteMessage(others: AnyRef, _) =>
-          failStage(new RuntimeException(s"unexpected type Point or annotated with Measurement required"))
-      }
+        .groupBy(im => (im.databaseName, im.retentionPolicy))
+        .map(wm => toBatchPoints(wm._1._1, wm._1._2, wm._2))
+        .foreach(influxDB.write)
 
   }
 
   final class InfluxDBMapperRecordLogic extends InfluxDBLogic {
 
-    protected var influxDBMapper: InfluxDBMapper = _
-
-    override def preStart(): Unit =
-      influxDBMapper = new InfluxDBMapper(influxDB)
+    private val mapperHelper: InfluxDBResultMapperHelper = new InfluxDBResultMapperHelper
 
     override protected def write(messages: immutable.Seq[InfluxDBWriteMessage[T, C]]): Unit =
-      messages.foreach {
-        case InfluxDBWriteMessage(typeMetric: Any, _) =>
-          influxDBMapper.save(typeMetric)
+      messages
+        .groupBy(im => {
+          (
+            im.databaseName match {
+              case Some(databaseName) => Some(databaseName)
+              case None => Some(mapperHelper.databaseName(im.point))
+            },
+            im.retentionPolicy match {
+              case Some(databaseName) => Some(databaseName)
+              case None => Some(mapperHelper.retentionPolicy(im.point))
+            }
+          )
+        })
+        .map(
+          wm =>
+            toBatchPoints(wm._1._1,
+                          wm._1._2,
+                          wm._2.map(im => im.withPoint(mapperHelper.convertModelToPoint(im.point).asInstanceOf[T])))
```

Comment:
`im => im` -> `_`

Reply:
Yes indeed I tried to use the underscore however im is needed for the 

```java
mapperHelper.convertModelToPoint(im.point).asInstanceOf[T])
```

Reply:
Ah, my apologies, missed that. 👍